tests/core/test_tracer.py

```diff
@@ -29,6 +29,25 @@ def test_variable_stats():
     assert_equals(tracer.usage_stats['C']['nb_requests'], 0)
 
 
+def test_log_format():
+    tracer = Tracer()
+    tracer.record_calculation_start("A", 2017)
+    tracer.record_calculation_start("B", 2017)
+    tracer.record_calculation_end("B", 2017, 1)
+    tracer.record_calculation_end("A", 2017, 2)
+
+    from io import StringIO
+    import contextlib
+
+    log = StringIO()
+    with contextlib.redirect_stdout(log):
+        tracer.print_computation_log()
```

Comment:
Just to avoid this awkwardness, it might be worth reworking the implementation to a more functional style; have a `generate_computation_log` return an array of appropriately indented lines, and unit test that more easily; and rewrite `print_computation_log` as the trivially correct

```
def print_computation_log():
  print(generate_computation_log.join("\n")
```

Not a very hard change, and it's a small but appreciable bit of improvement in code quality.

Relatedly I notice that the `print_trace` function in `tracers.py` seems unused anywhere in the entire https://github.com/openfisca/ org. Maybe remove it as dead code, if that doesn't break any test ?